### PR TITLE
adding extra keyword arguments to pass to the models constructor

### PR DESCRIPTION
**Pull request by pylover**

### CODE REVIEW

**Comment by dahlia** [comment]
Seems a good idea, but it still doesn’t allow to pass arguments if a constructor takes parameters named `size`/`mimetype`/`original`/etc.  I think it would be more flexible if it takes `args, kwargs` instead of `**kwargs`.

**Comment by pylover** [comment]
good idea.

**Comment by dahlia** [comment]
Could you rebase it on the current master?

**Comment by pylover** [comment]
here you go

File: sqlalchemy_imageattach/entity.py [hunk 1]
```diff
@@ -568,14 +578,18 @@ def from_blob(self, blob, store=current_store):
...
+        :type extra_args: :class:`list`
+        :param extra_kwargs: additional keyword arguments to pass to the model's constructor.
+        :type extra_kwargs: :class:`dict`
         :returns: the created image instance
         :rtype: :class:`Image`
 
         """
         data = io.BytesIO(blob)
-        return self.from_raw_file(data, store, original=True)
+        return self.from_raw_file(data, store, original=True, extra_args=extra_args, extra_kwargs=extra_kwargs)
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: Please [limit lines to a maximum 79 characters][1] e.g.:

```python
return self.from_raw_file(data, store,
                          original=True,
                          extra_args=extra_args,
                          extra_kwargs=extra_kwargs)
```

[1]: https://www.python.org/dev/peps/pep-0008/#maximum-line-length

File: sqlalchemy_imageattach/entity.py [hunk 2]
```diff
@@ -584,14 +598,20 @@ def from_file(self, file, store=current_store):
...
         :rtype: :class:`Image`
 
         """
+        if isinstance(file, cgi.FieldStorage):
+            file = file.file
         data = io.BytesIO()
         shutil.copyfileobj(file, data)
         data.seek(0)
-        return self.from_raw_file(data, store, original=True)
+        return self.from_raw_file(data, store, original=True, extra_args=extra_args, extra_kwargs=extra_kwargs)
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: This too.

File: sqlalchemy_imageattach/entity.py [hunk 3]
```diff
@@ -584,14 +598,20 @@ def from_file(self, file, store=current_store):
                       :data:`~sqlalchemy_imageattach.context.current_store`
                       by default
         :type store: :class:`~sqlalchemy_imageattach.store.Store`
+        :param extra_args: additional arguments to pass to the model's constructor.
+        :type extra_args: :class:`list`
+        :param extra_kwargs: additional keyword arguments to pass to the model's constructor.
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: This too.

File: sqlalchemy_imageattach/entity.py [hunk 4]
```diff
@@ -517,6 +517,10 @@ def from_raw_file(self, raw_file, store=current_store, size=None,
                          it is an original image or not.
                          defualt is ``True`` (meaning original)
         :type original: :class:`bool`
+        :param extra_args: additional arguments to pass to the model's constructor.
+        :type extra_args: :class:`list`
+        :param extra_kwargs: additional keyword arguments to pass to the model's constructor.
+        :type extra_kwargs: :class:`dict`
         :returns: the created image instance
         :rtype: :class:`Image`
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: To log changes, please add the following RST directive into the docstring:

```rst
.. versionadded:: 0.9.1
   The ``extra_args`` and ``extra_kwargs`` options.
```

File: sqlalchemy_imageattach/entity.py [hunk 5]
```diff
@@ -568,14 +578,18 @@ def from_blob(self, blob, store=current_store):
                       :data:`~sqlalchemy_imageattach.context.current_store`
                       by default
         :type store: :class:`~sqlalchemy_imageattach.store.Store`
+        :param extra_args: additional arguments to pass to the model's constructor.
+        :type extra_args: :class:`list`
+        :param extra_kwargs: additional keyword arguments to pass to the model's constructor.
+        :type extra_kwargs: :class:`dict`
         :returns: the created image instance
         :rtype: :class:`Image`
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: To log changes, please add the following RST directive into the docstring:

```rst
.. versionadded:: 0.9.1
   The ``extra_args`` and ``extra_kwargs`` options.
```

File: sqlalchemy_imageattach/entity.py [hunk 6]
```diff
@@ -584,14 +598,20 @@ def from_file(self, file, store=current_store):
...
```
Comment (dahlia) [inline sqlalchemy_imageattach/entity.py]: To log changes, please add the following RST directive into the docstring:

```rst
.. versionadded:: 0.9.1
   The ``extra_args`` and ``extra_kwargs`` options.
```

**Comment by dahlia** [comment]
Too many commits for this. :sweat_smile:  Could you rebase them?  I think it can be squashed to one commit.

**Comment by pylover** [comment]
of course. no problem. i like your library

**Comment by pylover** [comment]
done

**Comment by dahlia** [comment]
Still too many commits… please rebase them on the current master and squash them into one commit. :disappointed_relieved:

**Comment by pylover** [comment]
Just rebased

**Comment by dahlia** [comment]
Even more commits were made. Could you squash all these commits into **one single** commit?  Try utilizing `git reset --mixed`.

**Pull request closed without merging**